interview-questions.md

```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Most parents or guardians share their traits and properties with their children. Well! React carries this sharing and caring mentality my using props to send various properties of the parent class to its children classes or components. Props cannot be manipulated on the children class just displayed.
 
-  Researched answer:
+  Researched answer: Key words and tricky phrases!!! React certainly has some builtin tools to assist a developer. Let's look at props aka properties. `Props` is a keyword that allow properties to be shared between components. Child components normally pass data from the state using this technique. Props are immutable.
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: Device Object Model events are the changes made to the state of an object, which can be viewed on the UI. For instance, each letter that I typed on the computer is changing the state and therefore producing an even that can be seen on the screen.
 
-  Researched answer:
+  Researched answer:The webpage is loading. An impatient person is clicking the mouse. The image has finally loaded on the computer screen. Now the irritated person can finally input her user id and password. Yes, a common occurrence in a workspace. Additionally, these are also examples of DOM (Document Object Model) Events. Each event are signals or triggers to state that something is occurring or has occurred. As you can see, they are dependent on the browser or user interactions.
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: Who's down with OOP? If you are dealing with React, you better be. OOP or object-oriented programming is everything that its name leads you to think and more. You can perform many interactions for the user by setting code block based on manipulating objects and classes. OF course, all great things have to face competition---functional programming. If you are not OOPing, you are functioning your way into a useful webpage.
 
-  Researched answer:
+  Researched answer: Object-oriented programming uses classes and objects to manipulate data. Four main principles---encapsulation, abstraction, inheritance, and polymorphism---help OOP language to establish how the properties of a class or object will be viewed or manipulated. Functional programming has more freedom to be on any other code block not having to be tied to any classes or objects. Functions are used to produce new variables. No time for storage just serving outputs.
```

Comment:
Excellent comparison! React for all of its fun with class components actually does an amazing job with either OOP or FP. Lots of react Dev are moving to purely FP because it makes components smaller and more manageable (well if you follow the style guide it does).